ls1intum/Artemis
# [Feature] Reset SCA categories

### Checklist

### Motivation and Context


Adds missing functionality to restore the default configuration for static code analysis categories.

### Description

- Move REST request handler and service of reset test cases to the test case resource and service
- Add functionality to reset static code analysis categories

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create a programming exercise with SCA
4. Navigate to the grading page
5. Alter the test case configuration and save
6. Reset the test case configuration
--> All test cases should now have the weight 1, bonusMultiplier 1 and bonus 0
7. Select the tab for the SCA categories
8. Remember the settings e.g. take a screenshot
9. Alter the category configuration and save
10. Reset the categories
--> The initial category state should be restored

### Test Coverage
